src/understand-tidb/the-lifecycle-of-a-statement.md

```diff
@@ -1 +1,200 @@
 # The Lifecycle of a Statement
+
+## MySQL protocol package with command and statement string
+
+After connecting and getting authenticated, the server is in a [statement execution loop](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L938) until the client is disconnected.
+
+The [dispatch](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1111) function checks what kind of command was sent through the MySQL protocol and dispatches the matching function, like this snippet:
+```golang
+	switch cmd {
+	// ...
+	case mysql.ComQuit:
+		return io.EOF
+	case mysql.ComInitDB:
+		if err := cc.useDB(ctx, dataStr); err != nil {
+			return err
+		}
+		return cc.writeOK(ctx)
+	case mysql.ComQuery: // Most frequently used command.
+		return cc.handleQuery(ctx, dataStr)
+	// ...
+	}
+```
+Where [mysql.ComQuery](https://github.com/pingcap/parser/blob/d4a88481405f8c59d45fc0a9c38ee24d55b9bf49/mysql/const.go#L102) is routed to [handleQuery](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1633), which handles all different non-prepared statements (some commands like change database/schema or ping are handled directly in the dispatch function).
+
+TiDB keep the state between statements like sql_mode, transaction state etc. in the clientConn's [sessionctx.Context](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/sessionctx/context.go#L43) struct.
+
+## High level code for handling a query
+
+Real types and function names, but only high level for less distraction by too much details
+
+Further explanations below.
+
+```golang
+// handleQuery is the entry point for running client connection statements/queries
+func (cc *clientConn) handleQuery(ctx context.Context, sql string) (error) {
+	stmts, err := cc.ctx.Parse(ctx, sql)
+	// ...
+
+	for i, stmt := range stmts {
+		retryable, err = cc.handleStmt(ctx, stmt, ...)
+		// ...
+	}
+}
```

Comment:
Maybe briefly note that:

The MySQL protocol is synchronous, and the typical execution flow revolves around a client sending a single query, and receiving an optional result set.

As shown here; it is possible that a client might send multiple queries in one `mysql.ComQuery` call, in which case the `cc.ctx.Parse` will return multiple results. However; this is not a common occurence. By default, multiple statements in one  `mysql.ComQuery` call is disabled for security reasons. Clients must explicitly enable the `ClientMultiStatements` protocol feature.

Reply:
Thank you, I added a slightly modified version of your note.